lib/max_subarray.rb

```diff
@@ -1,8 +1,21 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 def max_sub_array(nums)
```

Comment:
The space complexity here is O(1) because you're not building another array etc.